### ISSUE TITLE
abbreviation difference from scispacy

### ISSUE THREAD

**Issue by danielkingai2** [erre-quadro/spikex#2]
Hi! scispacy developer here. Could you share what changes you made to our abbreviation detector? I am curious what issues you encountered/fixed (obviously not bothered at all that you based yours off of ours).

**Comment by paoloq** [comment]
Hi! Thanks for asking, I really love your work with scispacy, it's impressive.

Actually, I didn't fix any issue on your abbreviation detector, at least not on purpose.
Erre Quadro works on patents, which are very technical and heterogeneous documents, and I needed to have an our own abbreviation detector to be able to fix special cases as soon as encountered. 

As main differences, I added:
* squared brackets in matching short/long forms, together with a no brackets pattern
* detection more strict in some cases, forcing to match the first letter of a word
* forced to skip one word max in checking short and long forms

There may be others, but would be minor.

On next updates, I would like to better handle: 
* long/short forms punctuation separators (for example allowing dashes and not dots)
* long/short forms overlapping (one cancel other in global matching)

Of course I welcome any contribution or feedback, and thanks again for being inspiring.
